sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -375,6 +375,84 @@ class FilesElement extends Polymer.Element {
   }
 
   /**
+   * This is the entry point for file upload functionality. This is here to avoid using
+   * the very ugly <input> element for file uploads.
+   */
+  _altUpload() {
+    this.$.altFileUpload.click();
+  }
+
+  /**
+   * Gets called after the user selected one or more files from the upload modal.
+   * For each of the selected files, reads its contents, converts it to base64, then
```

Comment:
Do you need to read the file into memory in the client? Can you use the xhr upload API?